Refine sample kubernetes manifest file for simplicity and ease of use

semgrep-jamie opened the pull request:
As of v0.25.0 and greater, the requirements for fixed values within the `config.yaml` file for the broker are **much** lighter. Specifying the default publicKey and allowedIps isn't necessary. The addition of the `--deployment-id` flag in #72 means that the broker can be started with very minimal configuration needs, and will load the default config from https://semgrep.dev/api/broker/16/default-config instead (where `16` is the deployment/org id).

Our docs overall (both this repo's README.md and https://semgrep.dev/docs/semgrep-ci/network-broker) don't reflect this potential simplicity and overall, need an update and reconciliation so that the same info is available in both.

Technically, this example could be _further_ simplified, and the config.yaml file completely removed. Only the `inbound:wireguard:privateKey` value is required. I opted to keep config.yaml present as it's likely users will want to tune the allowlist and SCM configuration.

**A future enhancement:** It's unclear to me if the github:* map is _even required_ for SMS functionality. We should do some lab testing to confirm or deny this. That could unlock further simplification.
edit: lab testing confirmed that at least the scm:baseURL: map is required for SMS to work via broker. But not the token: key.

semgrep-jamie committed:
Simplify & clarify sample kubernetes manifest

semgrep-jamie committed:
remove unnecessary inbound:wireguard map and show an allowlist & github config instead.

semgrep-jamie committed:
avoid including GITHUB_PAT in config.yaml and instead mount it in the secret volume

semgrep-jamie committed:
Remove inbound:github:token portion as it's not required

semgrep-jamie commented:
shoutout to @stuartcmehrens for helping confirm that the `inbound:github:token` (or `inbound:gitlab:token`) dict **is not required**. It's cleanest to configure the token within semgrep.dev settings only.

Updated in [50964d4](https://github.com/semgrep/semgrep-network-broker/pull/155/commits/50964d42b95ec72445f77af69092575a0f52cec1).

kyle-semgrep reviewed (approved):
Looks good Jamie!  Thank you!

tpetr commented on kubernetes.yaml:
```diff
@@ -4,21 +4,13 @@ metadata:
-        - url: http://example.com/*
-          methods:
-          - GET
+      allowlist: [] # Optional. Empty list means all traffic outbound from the broker is permitted
```
Uber-nit but can you add some wording here clarifying that `allowlist` defaults to `[]` if not set?

semgrep-jamie replied:
Done in [c88f7aa](https://github.com/semgrep/semgrep-network-broker/pull/155/commits/c88f7aa767bd759039e3f1e43bc12b93cf9e2854)

tpetr commented on kubernetes.yaml:
```diff
@@ -4,21 +4,13 @@ metadata:
-          - GET
+      allowlist: [] # Optional. Empty list means all traffic outbound from the broker is permitted
+      github:
+        baseURL: https://GITHUB_BASE_URL/api/v3
```
2 nits:
1. `GITHUB_BASE_URL` is technically serving as the placeholder for the ghe hostname, so might be good to be super explicit and do something like `GHES_HOSTNAME` instead
2. Might want to either a) add a comment here or b) an additional sentence on line 8 reminding folks to replace this placeholder with their GHES info

semgrep-jamie replied:
Addressed both in [c88f7aa](https://github.com/semgrep/semgrep-network-broker/pull/155/commits/c88f7aa767bd759039e3f1e43bc12b93cf9e2854), opted for 2a.

tpetr commented on kubernetes.yaml:
```diff
@@ -37,11 +29,15 @@ spec:
-      - name: gateway
-        image: image-goes-here
+      - name: broker
+        image: ghcr.io/semgrep/semgrep-network-broker:v0.34.0 # Update to desired tag version from https://github.com/semgrep/semgrep-network-broker/tags
```
I wonder if we should pin this to `:latest` so that we don't have to worry about folks accidentally staying on an old version (and then update the comment to say something like "or pin to desired version from ...")? OTOH, it's generally sorta frowned upon to use `:latest` since it means the broker could upgrade without people realizing. Wanted to flag but up to you re: how to proceed

semgrep-jamie replied:
Good call out - I had the same debate. I think `:latest` will be more durable and doesnt create the need to regularly update this sample manifest.

Added a comment in [c88f7aa](https://github.com/semgrep/semgrep-network-broker/pull/155/commits/c88f7aa767bd759039e3f1e43bc12b93cf9e2854) clarifying that they can optionally change `:latest` to a specific tagged version.

tpetr reviewed (approved):
A few nits but great overall, thanks!

semgrep-jamie committed:
Address tom's feedback re:
  - GitHub base url placeholder
  - clarifying default behaviour of allowlist when unset.
  - tag version of manifest image

semgrep-jamie committed:
remove trailing whitespace from L34
